shared/js/smart-screen/card_manager.js

```diff
@@ -126,7 +126,8 @@
         return that._cardStore.saveData('cardList', cardEntries);
       }).then(function() {
         that._asyncSemaphore.p();
-        if (options && options.cleanEmptyFolder) {
+        if (options && options.cleanEmptyFolder &&
+          emptyFolderIndices.length > 0) {
           that.fire('card-removed', emptyFolderIndices);
```

Comment:
Should ONLY fire card-removed event when there are empty folders.
Should NOT fire card-removed event when there is no empty folders.